Feature/card 5 create class enums

andrefajardo opened the pull request:
1 - Create - Enums "StatusMatricula" e "StatusNivel" em atenção ao requisito do Card n° 5, do projeto M1S06.

2 - Update - Code refactoring for optimization in Enums "StatusMatricula" and "StatusNivel" according to card 5 requirements. (M1S06).

3 - Update - Code refactoring for optimization in Enums "StatusMatricula" and "StatusNivel" according to card 5 requirements. (M1S06).

4 - Fix - Fix for .idea issue.

5 - Update  Code refactoring for optimization in Enums "StatusMatricula" and "StatusNivel" according to card 5 requirements.
Adaptation to the correct language (M1S06).

6 - Created the "models" package to contain usage models.

6.1 - Creating a class to test the values() and valueOf() methods.

7 - Implementation of a method to obtain the value from the index.

8 - Function calls showing use cases.


<Resp: José André - Ref #Card-5>

andrefajardo committed:
Create - M1S06Ex5

Create - Enums "StatusMatricula" e "StatusNivel" em atenção ao requisito do Card n° 5, do projeto M1S06.

<Resp: José André - Ref #Card>

andrefajardo committed:
Merge branch 'develop' of https://github.com/andrewmatheus/fs-challenge-squad-school-admin into feature/card-5-create-class-enums

andrefajardo committed:
Update arquives StatusMatricula.java and StatusNivel.java

Code refactoring for optimization in Enums "StatusMatricula" and "StatusNivel" according to card 5 requirements. (M1S06).

<Resp: José André - Ref #Card-5>

andrefajardo committed:
Update - Fix for .idea issue.

Fix for .idea issue.

<Resp: José André - Ref #Erro>

andrefajardo committed:
Update arquives StatusMatricula.java and StatusNivel.java

Code refactoring for optimization in Enums "StatusMatricula" and "StatusNivel" according to card 5 requirements.
Adaptation to the correct language (M1S06).

<Resp: José André - Ref #Card-5>

andrefajardo committed:
Create TestsEnum.java

Created the "models" package to contain usage models;
Criação de uma classe para testar os métodos values() e valueOf()

<Resp: José André - Ref #Card-5>

andrefajardo committed:
Updata  -  Functionality implementation

Implementation of a method to obtain the value from the index.

<Resp: José André - Ref #Card-5>

andrefajardo committed:
Update Main.java

Function calls showing use cases.

<Resp: José André - Ref #Card-5>

micheldrv commented on src/enums/StatusLevel.java:
```diff
@@ -0,0 +1,30 @@
+public enum StatusLevel {
+    BEGINNER(1),
+    EXPERIENCED(2),
+    ADVANCED(3);
```
Só uma sugestão, acho que ficaria menos ambíguo se esse enum fosse renomeado para algo como EmployeeLevel.

andrefajardo replied:
Achei a ideia excelente! Muito mais claro e inteligível.

andrefajardo committed:
Merge branch 'develop' into feature/card-5-create-class-enums

andrefajardo committed:
Refactor - Change of nomenclature

Changing file names of arquives StatusMatricula.java and StatusNivel.java and adjusting calls.

<Resp: José André - Ref #Card-5>

andrefajardo committed:
Refactor - Adjustments to calls and imports

Just adjustments to calls and imports.

<Resp: José André - Ref #Card-5>

andrefajardo committed:
Merge branch 'feature/card-5-create-class-enums' of https://github.com/andrewmatheus/fs-challenge-squad-school-admin into feature/card-5-create-class-enums
